public/app/mock-api-utils.ts

```diff
@@ -0,0 +1,37 @@
+import { sendAppNotification } from './core/copy/appNotification';
+import { AppNotificationSeverity } from './types';
+
+export const STORAGE_MOCK_API_KEY = 'grafana.dev.mockApi';
+
+export const currentMockApiState = () => {
+  return window.localStorage.getItem(STORAGE_MOCK_API_KEY) === 'true';
+};
+
+export const toggleMockApiAndReload = () => {
+  const currentState = currentMockApiState();
+  window.localStorage.setItem(STORAGE_MOCK_API_KEY, String(!currentState));
+  sendAppNotification(`Toggling Mock API`, 'Reloading...', AppNotificationSeverity.Info);
+  setTimeout(() => {
+    window.location.reload();
+  }, 200);
+};
+
+export const potentiallySetupMockApi = async () => {
+  const mockApiEnabled = currentMockApiState();
+  if (process.env.NODE_ENV === 'development' && mockApiEnabled) {
+    const { default: worker } = await import('test/mock-api/worker');
+
+    worker.start({ onUnhandledRequest: 'bypass' });
+  }
+  return Promise.resolve();
```

Comment:
does this need to return `Promise.resolve()`? it looks like it can can have a plain return or not even need one, the function is async so it gets wrapped either way.